### ISSUE TITLE
support reading externalS3 config from environment variables. [Feature]: 

### ISSUE THREAD

**Issue by bibilins** [milvus-io/milvus#26571]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

now external tool configs currently reads passwords from configs/milvus.yaml, and we hope to support reading them from environment variables。 accesskey && secretKey of externalS3 are storaged in k8s secret，and we want to read these config from k8s secret。

### Describe the solution you'd like.

support reading externalS3 config from environment variables

### Describe an alternate solution.

no

### Anything else? (Additional Context)

no

**Comment by xiaofan-luan** [comment]
so you are saying reading a system env like 
milvus.minio.accessKeyID as the config?

**Comment by bibilins** [comment]
> so you are saying reading a system env like milvus.minio.accessKeyID as the config?

like `MINIO_ACCESS_KEY` && `MINIO_SECRET_KEY`

**Comment by weiZhenkun** [comment]
> so you are saying reading a system env like milvus.minio.accessKeyID as the config?

Yes, We are also need this feature. the real UserName or Passwrod or Token are saved in K8S secret, and the milvus_value.yaml should only save their K8S secret keys. All the secret for extenal S3, extennal Kafka should be apply this feature.

**Comment by xiaofan-luan** [comment]
/assign @haorenfsa 

coudl you help on this?

**Comment by haorenfsa** [comment]
yes, there's not much work to support these.
By the way, you can also try using the `minio.useIAM` config together with [AWS: Configuring Pods to use a Kubernetes service account](https://docs.aws.amazon.com/eks/latest/userguide/pod-configuration.html) Or [GCP: Allow Pods to authenticate to Google Cloud APIs using Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) to avoid expose any secret keys in the kubernetes cluster.

**Comment by bibilins** [comment]
but our environment is a private deployment, please support the use of environment variables

**Comment by Mawiszus** [comment]
I would like to add to this issue, as I also want to use environment variables to define the Minio Credentials.

I am using the standalone version and tried to set `MINIO_ACCESS_KEY` and `MINIO_SECRET_KEY` with the docker compose file, but it only works if I also change the corresponding settings in the config yaml file to the explicit secrets, which I would like to get around.

Looking at the docs [here](https://milvus.io/docs/configure_minio.md) for `minio.accessKeyID`, it says:
> This configuration must be set identical to the environment variable MINIO_ACCESS_KEY, which is necessary for starting MinIO or S3.

Have I been using them wrong? Otherwise, why have both `MINIO_ACCESS_KEY` and `minio.accessKeyID` if they need to be identical?

**Comment by haorenfsa** [comment]
@Mawiszus  setting the environment vars should be enough, it overwrites the  minio.accessKeyID

**Comment by haorenfsa** [comment]
The description in doc is a bit confusing, we'll update it later.

**Comment by RakeshRaj97** [comment]
Any updates on this?

**Comment by haorenfsa** [comment]
For milvus-helm, you need to add following configs for each of milvus component:
```
dataNode:
  extraEnv:
  - name: MINIO_ACCESS_KEY
    valueFrom:
      secretKeyRef:
        key: accesskey
        name: youSecretName
  - name: MINIO_SECRET_KEY
    valueFrom:
      secretKeyRef:
        key: secretkey
        name: youSecretName
```

**Comment by iamcodingcat** [comment]
@xiaofan-luan Hi
Currently, the Helm chart manifests used to install Milvus—such as the values.yaml file—are being managed in a GitOps manner. As you know, the GitOps approach involves version-controlling the manifest source code in a GitHub repository.

However, when looking at the currently supported Milvus Helm charts, if you’re using an external storage backend like AWS S3, you have to specify AWS account credentials so that Milvus can access the S3 bucket. The problem is that these credentials must be written in plain text directly into the source code—hardcoded and not even encrypted. When this kind of sensitive information is managed in plain text at the source code level and committed to a GitHub repository, it poses a significant security risk.

Personally, I strongly believe that from a security perspective, credentials should be injected in the form of Kubernetes Secrets. I’m curious if there are any plans to include this functionality in the upcoming release milestones.

**Comment by younghun-jo-levit-com** [comment]
@haorenfsa your answer is applied to S3 sotrage replacing with MinIO, too?

> For milvus-helm, you need to add following configs for each of milvus component:
> 
> ```
> dataNode:
>   extraEnv:
>   - name: MINIO_ACCESS_KEY
>     valueFrom:
>       secretKeyRef:
>         key: accesskey
>         name: <yourSecretName>
>   - name: MINIO_SECRET_KEY
>     valueFrom:
>       secretKeyRef:
>         key: secretkey
>         name: <yourSecretName>
> ```

**Comment by haorenfsa** [comment]
@iamcodingcat Yes, it should be applied to S3, too.  But I pasted the wrong env name, should be

`MINIO_ACCESS_KEY_ID` & `MINIO_SECRET_ACCESS_KEY`
![Image](https://github.com/user-attachments/assets/d4e919da-5535-4b19-a93a-c1bed44937b9)

**Comment by iamcodingcat** [comment]
@haorenfsa thanks. I am waiting for your answer. please!

**Comment by iamcodingcat** [comment]
@haorenfsa Really thanks for fast reply! I succeed in injecting s3 credentials using secret. 
If I fail to doing this, I am trying to use pod-identity of aws(ref: [this issue](https://github.com/milvus-io/milvus/issues/34910))

**Comment by haorenfsa** [comment]
Yes, pod identity is recommended, so that you won't need an AK and has no worries about security.

**Comment by marco210** [comment]
same issue, waiting for your feedback
